### PR TITLE
Landing page UI

### LINKED ISSUE #43: FE: Landing Page
Create the UI for the Homepage/Landing Page

### PR DESCRIPTION
**Pull request by samsoncsyu7777**
### What this PR does (required):
-conditional narBar by putting it inside <switch>
-construct Landing Page and style it
-construct FindSitter Form
-use datePicker library to pick the date for the input of the form

### Screenshots / Videos (front-end only):
Screenshot 2021-11-18 at 10 25 10 PM

### Any information needed to test this feature (required):
- npm start
- http://localhost:3000/

### CODE REVIEW

**Review by sundayezeilo — COMMENTED** [review]
UI Looks good!
:+1:

**Comment by samsoncsyu7777** [comment]
> UI Looks good! 👍

Thank you.

File: client/src/components/NavBar/AuthBars/useStyles.ts [hunk 1]
```diff
@@ -1,6 +1,16 @@
 import { makeStyles } from '@material-ui/core/styles';
 
 const useStyles = makeStyles(() => ({
+  accAside: {
+    textDecoration: 'underline',
+    textTransform: 'uppercase',
```
Comment (rajivtitus) [inline client/src/components/NavBar/AuthBars/useStyles.ts:6]: 💯

File: client/src/themes/theme.ts [hunk 2]
```diff
@@ -10,6 +10,12 @@ export const theme = createMuiTheme({
     body1: {
       color: '#f14140',
     },
+    h2: {
```
Comment (rajivtitus) [inline client/src/themes/theme.ts:13]: Great work using MUI's theme the way it's supposed to be used 👍🏽

File: client/src/components/NavBar/NavBar.tsx [hunk 3]
```diff
@@ -6,18 +7,29 @@ import CircularProgress from '@material-ui/core/CircularProgress';
...
-  const classes = useStyles();
+interface Props {
+  children?: React.ReactNode;
+}
 
+const NavBar = ({ children }: Props): JSX.Element => {
+  const [isLandingPage, setIsLandingPage] = useState<boolean>(false);
+  const classes = useStyles();
   const { loggedInUser } = useAuth();
   const { initSocket } = useSocket();
```
Comment (rajivtitus) [inline client/src/components/NavBar/NavBar.tsx]: You can remove this line of code and the useEffect below which initializes the socket, we don't want to initialize the socket within the Navbar :)

File: client/src/components/NavBar/NavBar.tsx [hunk 4]
```diff
@@ -26,13 +38,20 @@ const NavBar = (): JSX.Element => {
...
-        {loggedInUser ? <LoggedInBar /> : <LoggedOutBar />}
-      </ToolBar>
-    </AppBar>
+    <Grid>
+      <AppBar
+        elevation={history.location.pathname === '/' ? 0 : 6}
+        className={isLandingPage ? classes.landingPageBar : classes.appbar}
+        position="absolute"
+      >
+        <CssBaseline />
```
Comment (rajivtitus) [inline client/src/components/NavBar/NavBar.tsx]: Please remove the `CssBaseline`component as is a global reset and should be included only once, ideally in the App.tsx

File: client/src/components/NavBar/NavBar.tsx [hunk 5]
```diff
@@ -26,13 +38,20 @@ const NavBar = (): JSX.Element => {
...
+        <ToolBar className={classes.toolbar}>
+          <img src={logo} alt="logo" />
+          {loggedInUser ? <LoggedInBar /> : <LoggedOutBar isLandingPage={isLandingPage} />}
```
Comment (rajivtitus) [inline client/src/components/NavBar/NavBar.tsx:43]: 👍🏽

File: client/src/pages/Landing/Landing.tsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  const handleSubmit = (
+    { address, startDate, endDate }: { address: string; startDate: Date; endDate: Date },
+    { setSubmitting }: FormikHelpers<{ address: string; startDate: Date; endDate: Date }>,
+  ) => {
+    //construct this function in the next ticket
+  };
+
+  return (
+    <Grid container component="main" className={classes.root}>
+      <CssBaseline />
```
Comment (rajivtitus) [inline client/src/pages/Landing/Landing.tsx]: Same as comment above regarding the CssBaseline component

File: client/src/pages/Landing/Landing.tsx [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+      <Grid className={classes.column}>
```
Comment (rajivtitus) [inline client/src/pages/Landing/Landing.tsx]: The Grid component has two types - item/container. Please specify the appropriate type here and throughout :)

File: client/src/pages/Landing/LandingForm/LandingForm.tsx [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+      endDate: Date;
+    }>,
+  ) => void;
+}
+
+export default function FindSitter({ handleSubmit }: Props): JSX.Element {
+  const classes = useStyles();
+
+  return (
+    <Formik
```
Comment (rajivtitus) [inline client/src/pages/Landing/LandingForm/LandingForm.tsx:38]: 💯

File: client/src/pages/Landing/useStyles.ts [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+  },
+  column: {
+    width: '50vw',
+  },
+  dogs: {
+    width: '100%',
+    height: '100%',
+  },
+  authWrapper: {
+    display: 'flex',
```
Comment (rajivtitus) [inline client/src/pages/Landing/useStyles.ts]: You can apply any significant styles to the Box component directly

File: client/src/pages/Landing/useStyles.ts [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+    '& .MuiInput-underline:before': {
+      borderBottom: '1.2px solid rgba(0, 0, 0, 0.2)',
+    },
+    minWidth: '100vw',
+    display: 'flex',
+  },
+  column: {
+    width: '50vw',
+  },
+  dogs: {
```
Comment (rajivtitus) [inline client/src/pages/Landing/useStyles.ts:15]: Optional: You can add `objectFit: 'cover'` to images to preserve the aspect ratio 😉

**Review by rajivtitus — COMMENTED** [review]
Excellent work with this! Just a few comments from my end, please have a look and adderss, thanks!

File: client/src/components/NavBar/NavBar.tsx (hunk 3 above)
Comment (samsoncsyu7777) [reply client/src/components/NavBar/NavBar.tsx]: Yes, it's true. Done

File: client/src/components/NavBar/AuthBars/useStyles.ts (hunk 1 above)
Comment (samsoncsyu7777) [reply client/src/components/NavBar/AuthBars/useStyles.ts:6]: Thanks.

File: client/src/themes/theme.ts (hunk 2 above)
Comment (samsoncsyu7777) [reply client/src/themes/theme.ts:13]: Thank you.

File: client/src/components/NavBar/NavBar.tsx (hunk 4 above)
Comment (samsoncsyu7777) [reply client/src/components/NavBar/NavBar.tsx]: I used the code from another component and forgot to check it. Done.

File: client/src/components/NavBar/NavBar.tsx (hunk 5 above)
Comment (samsoncsyu7777) [reply client/src/components/NavBar/NavBar.tsx:43]: Thanks.

File: client/src/pages/Landing/Landing.tsx (hunk 6 above)
Comment (samsoncsyu7777) [reply client/src/pages/Landing/Landing.tsx]: :) Done.

File: client/src/pages/Landing/Landing.tsx (hunk 7 above)
Comment (samsoncsyu7777) [reply client/src/pages/Landing/Landing.tsx]: I was told that Grid has item property by default last year. Today, I found that both item and container are false by default from MUI API docs. Actually, I use item here.

File: client/src/pages/Landing/LandingForm/LandingForm.tsx (hunk 8 above)
Comment (samsoncsyu7777) [reply client/src/pages/Landing/LandingForm/LandingForm.tsx:38]: Thanks.

File: client/src/pages/Landing/useStyles.ts (hunk 9 above)
Comment (samsoncsyu7777) [reply client/src/pages/Landing/useStyles.ts]: Thanks for your reminder. Done.

File: client/src/pages/Landing/useStyles.ts (hunk 10 above)
Comment (samsoncsyu7777) [reply client/src/pages/Landing/useStyles.ts:15]: Haha, too hurry and missed it. Thank you. Fixed.

**Pull request merged**